core/Pos/Core/Slotting/Timestamp.hs

```diff
@@ -60,8 +68,21 @@ diffTimestamp t1 t2 = getTimestamp t1 - getTimestamp t2
 addMicrosecondsToTimestamp :: Microsecond -> Timestamp -> Timestamp
 addMicrosecondsToTimestamp m t = Timestamp { getTimestamp = (getTimestamp t) + m }
 
+-- | Lens to convert timestamp to fractional seconds and vice versa.
+timestampSeconds
+    :: (Eq a, RealFrac a)
+    => Iso' Timestamp a
+timestampSeconds = _Timestamp . iso fromIntegral round . dividing 1e6
```

Comment:
Am I the only one who finds the usage of the term `second` without the quantifying precision (as in `micro`) misleading? As we are dividing by `1e6`, should we call this `timestampMicroseconds`?

Reply:
As to me, it should be `second`, because
* looking at `timestampMicroseconds`, I don't see at glance which measure it returns time in. Maybe it returns time in `weeks` with up to `microseconds` precision :smile:
Perhaps more qualified name would be more telling.
* it actually returns time with precision which I'd expect when converting from `Timestamp`, i.e, exactly the same `up to mcs` as in `Timestamp`.

What do you think?

Reply:
Uhm, not totally convinced about the first point, I think 😁 `microseconds` is not a subjective time of measure, imho. I sort of see what you are saying, but for the _principle of least surprise_ `timestampMicroseconds` does exactly what it says in the title: it returns a timestamp expressed in microseconds. I'm curious to hear what @parsonsmatt thinks here 😉

Reply:
It's hard to me to agree, since if I had `t :: Timestamp` corresponding to `12.3 sec` (`Timestamp 12300000` inside), then this "function", applied to `t`, would return `12.3` and that would mean `12.3 sec`, not `12.3 mcs` :slightly_smiling_face: 
Anyway thanks for involving another couple of eyes into it, hope we'll reach a consensus :eyes:

Reply:
Yeah, that `Timestamp` and lens coupling is a bit tricky, perhaps I did a wrong thing when didn't insert an explaining example in haddock.